UI/src/view/PointAssetLayer.js

```diff
@@ -0,0 +1,230 @@
+          withDeactivatedSelectControl(function() {
+            me.removeLayerFeatures();
+          });
+          var features = _.map(assets, createFeature);
+          vectorLayer.addFeatures(features);
+          applySelection();
+        }
+      });
+    };
+
+    this.removeLayerFeatures = function() {
+      vectorLayer.removeAllFeatures();
+    };
+
+    function applySelection() {
+      if (selectedAsset.exists()) {
+        withoutOnSelect(function() {
+          var feature = _.find(vectorLayer.features, function(feature) { return selectedAsset.isSelected(feature.attributes); });
+          if (feature) {
+            me.selectControl.select(feature);
+          }
+        });
+      }
+    }
+
+    function withDeactivatedSelectControl(f) {
+      var isActive = me.selectControl.active;
+      if (isActive) {
+        me.selectControl.deactivate();
+        f();
+        me.selectControl.activate();
+      } else {
+        f();
+      }
+    }
+
+    function withoutOnSelect(f) {
+      me.selectControl.onSelect = function() {};
+      f();
+      me.selectControl.onSelect = pointAssetOnSelect;
+    }
+
+    function withoutOnUnselect(f) {
+      me.selectControl.onUnselect = function() {};
+      f();
+      me.selectControl.onUnselect = pointAssetOnUnselect;
+    }
+
+    this.layerStarted = function(eventListener) {
+      bindEvents(eventListener);
+    };
+
+    function toggleMode(readOnly) {
+      if(readOnly){
+        dragControl.deactivate();
+      } else {
+        dragControl.activate();
+      }
+    }
+
+    function bindEvents(eventListener) {
+      eventListener.listenTo(eventbus, 'map:clicked', handleMapClick);
+      eventListener.listenTo(eventbus, 'pedestrianCrossing:saved pedestrianCrossing:cancelled', handleSavedOrCancelled);
+      eventListener.listenTo(eventbus, 'pedestrianCrossing:creationCancelled', handleCreationCancelled);
+      eventListener.listenTo(eventbus, 'pedestrianCrossing:selected', handleSelected);
+      eventListener.listenTo(eventbus, 'pedestrianCrossing:unselected', handleUnSelected);
+      eventListener.listenTo(eventbus, 'pedestrianCrossing:changed', handleChanged);
+      eventListener.listenTo(eventbus, 'application:readOnly', toggleMode);
+    }
+
+    function handleCreationCancelled() {
+      withoutOnUnselect(function() {
+        me.selectControl.unselectAll();
+      });
+      mapOverlay.hide();
+      vectorLayer.styleMap = style.browsing;
+      me.refreshView();
+    }
+
+    function handleSelected() {
+      vectorLayer.styleMap = style.selection;
+      applySelection();
+      vectorLayer.redraw();
+    }
+
+    function handleSavedOrCancelled() {
+      mapOverlay.hide();
+      me.activateSelection();
+      me.refreshView();
+    }
+
+    function handleChanged() {
+      me.deactivateSelection();
+    }
+
+    function handleMapClick(coordinates) {
+      if (applicationModel.getSelectedTool() === 'Add' && zoomlevels.isInAssetZoomLevel(map.getZoom())) {
```

Comment:
It'd be nice if layers didn't have to care about zoom in click handling themselves.